src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java

```diff
@@ -292,16 +298,20 @@ private void switchTo(@Nullable SonarQubeServer server) {
   }
 
   private void setStatus(ConnectedSonarLintEngine.State state) {
-    ApplicationManager.getApplication().assertIsDispatchThread();
-
+    ConnectedSonarLintEngine.State currentState = engine.getState();
     StringBuilder builder = new StringBuilder();
-    switch (state) {
+
+    switch (currentState) {
       case NEVER_UPDATED:
         builder.append("never updated");
         break;
       case UPDATED:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)